src/OSSupport/GetAddressInfoError.h

```diff
@@ -0,0 +1,29 @@
+#pragma once
+
+
+
+/** Returns the readable form of a getaddressinfo type error code */
+inline AString ErrorString(int a_ErrorCode)
+{
+	// Note gai_strerror is not threadsafe on windows
+	#ifdef _WIN32
+		AString ErrorStr(GAI_STRERROR_BUFFER_SIZE, '\0');
+
+		int MsgLen = FormatMessageA(
+			FORMAT_MESSAGE_FROM_SYSTEM |
+			FORMAT_MESSAGE_IGNORE_INSERTS |
+			FORMAT_MESSAGE_MAX_WIDTH_MASK,
+			nullptr,
+			a_ErrorCode,
+			MAKELANGID(LANG_NEUTRAL, SUBLANG_DEFAULT),
+			&ErrorStr[0],
```

Comment:
This is a hack that is not guaranteed to be safe. Use a char buffer of (moderately large) fixed size instead.

I know we use this trick on a few places, but we try to phase it out as much as possible.